blocks/math.js

```diff
@@ -82,12 +82,19 @@ Blockly.Blocks['math_positive_number'] = {
    * @this Blockly.Block
    */
   init: function() {
-    this.setHelpUrl(Blockly.Msg.MATH_NUMBER_HELPURL);
-    this.setColour(Blockly.Blocks.math.HUE);
-    this.appendDummyInput()
-        .appendField(new Blockly.FieldNumber('0', 0, Infinity), 'NUM');
-    this.setOutput(true, 'Number');
-    this.setOutputShape(Blockly.OUTPUT_SHAPE_ROUND);
+    this.jsonInit({
+      "message0": "%1",
+      "args0": [
+        {
+          "type": "field_number",
+          "name": "NUM",
+          "min": 0
```

Comment:
Hmm, yeah. we'll have to think about the layering blocks, but it might make sense to have the signed int for backwards compatibility.

Graphic effects look to me like they can be affected by decimal values - try "set pixelate effect to 0.1" and then "set pixelate effect to 0.5".

Pen colors is an interesting example - I can't even really see the difference between colors until you start getting to increments of more than 1. So, maybe we should disallow the decimal values. But, I might argue that:
<img width="190" alt="screen shot 2016-07-10 at 11 43 04 am" src="https://cloud.githubusercontent.com/assets/120403/16714517/81d8b39e-4693-11e6-8069-3055fe13f406.png">
should be equivalent to:
<img width="165" alt="screen shot 2016-07-10 at 11 43 15 am" src="https://cloud.githubusercontent.com/assets/120403/16714520/86d0e65a-4693-11e6-840a-934b42c6a0eb.png">